src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/MultipartFormDataRequestContentProvider.cs

```diff
@@ -189,34 +180,44 @@ private Method BuildAddMethod<T>()
-                    contentExpression = New.Instance(typeof(StringContent), boolVariable);
-                    break;
-                case Type itype when itype == typeof(int):
-                case Type ftype when ftype == typeof(float):
-                case Type ltype when ltype == typeof(long):
-                case Type dtype when dtype == typeof(double):
-                case Type detype when detype == typeof(decimal):
-                    ValueExpression invariantCulture = new MemberExpression(typeof(CultureInfo), nameof(CultureInfo.InvariantCulture));
-                    var invariantCultureValue = new InvokeInstanceMethodExpression(contentParam, nameof(Int32.ToString), new[] {Literal("G"), invariantCulture }, null, false);
-                    valueDelareStatement = Declare(typeof(string), "value", invariantCultureValue, out TypedValueExpression variable);
-                    contentExpression = New.Instance(typeof(StringContent), variable);
-                    break;
-                case Type bdtype when bdtype == typeof(BinaryData):
-                    contentExpression = New.Instance(typeof(ByteArrayContent), new BinaryDataExpression(contentParam).ToArray());
-                    break;
+                var boolToStringValue = new TernaryConditionalOperator(contentParam, Literal("true"), Literal("false"));
+                valueDelareStatement = Declare(typeof(string), "value", boolToStringValue, out TypedValueExpression boolVariable);
+                contentExpression = New.Instance(typeof(StringContent), boolVariable);
+            }
+            else if (contentParamType == typeof(int)
+                || contentParamType == typeof(float)
+                || contentParamType == typeof(long)
+                || contentParamType == typeof(double)
+                || contentParamType == typeof(decimal))
+            {
+                ValueExpression invariantCulture = new MemberExpression(typeof(CultureInfo), nameof(CultureInfo.InvariantCulture));
+                var invariantCultureValue = new InvokeInstanceMethodExpression(contentParam, nameof(Int32.ToString), new[] { Literal("G"), invariantCulture }, null, false);
+                valueDelareStatement = Declare(typeof(string), "value", invariantCultureValue, out TypedValueExpression variable);
+                contentExpression = New.Instance(typeof(StringContent), variable);
+            }
+            else if (contentParamType == typeof(BinaryData))
+            {
+                contentExpression = New.Instance(typeof(ByteArrayContent), new BinaryDataExpression(contentParam).ToArray());
+            }
+            else if (contentParamType == typeof(byte[]))
+            {
+                contentExpression = New.Instance(typeof(ByteArrayContent), contentParam);
+            }
+            else
+            {
+                throw new Exception($"Not supported type {contentParamType}");
```

Comment:
if the message is "Not supported" how about we just throw an UnsupportedException?